### ISSUE TITLE
Request to support C language in Datahub

### ISSUE THREAD

**Issue by balajialg** [berkeley-dsep-infra/hubs#4930]
# Summary

Got a request from an instructor who is interested in using Python and C programming language in the same course. There are C kernels like [Jupyter C Kernel](https://github.com/brendan-rius/jupyter-c-kernel) and [C Kernel](https://github.com/adamtuft/c-kernel) that we could potentially explore. Can we install the C kernels in our image?

Inputs from the instructor below,

> I'm designing a course intended for biology majors titled, "Programming for biologists." My goal is to teach basic programming. The course will be language agnostic. That said, I will focus on two languages: C early in the semester and Python later in the semester.
> 
> I am interested in having a web-based interface through which the students can write and compile in both languages. I'm aware that we have excellent support for Jupyter Notebook on campus, and I suppose that would be fine, but I'm trying to minimize the angst students experience with compiling or running the code they write, so one unified interface would be best
>  
> John Huelsenbeck

**Comment by ryanlovett** [comment]
I'm not well informed about these, but there is also QuantStack's Xeus-based C++ kernel [xeus-cling](https://github.com/jupyter-xeus/xeus-cling) which supports C.
<img width="1092" alt="Screenshot 2023-08-24 at 2 15 46 PM" src="https://github.com/berkeley-dsep-infra/datahub/assets/10248125/910fd2d1-d00c-4b96-9354-37880487ba96">

**Comment by balajialg** [comment]
@ryanlovett Just played with a notebook based on this kernel in [binder](https://notebooks.gesis.org/binder/jupyter/user/jupyter-xeus-xeus-cling-ie5oxfcu/notebooks/notebooks/xcpp.ipynb). Super cool :)

Tried enabling it in Datahub by doing the following steps, it didn't work for me,

- Launched terminal and the following commands,
- Ran `mamba create -n cling`
- Ran `source activate cling`
- Ran `mamba install xeus-cling -c conda-forge`
- Launched a new notebook and checked available kernels - didn't work
- Restarted the kernel in an active notebook - didn't work

C kernel is not listed amongst the available kernels. Launcher also did not show an icon for this new kernel. Is it possible to install this kernel on the user server side or it needs to get added to environment.yml?

**Comment by ryanlovett** [comment]
@balajialg Yes, I believe so. In that example you're installing the kernel into a separate conda environment so you'd need to also create a kernel which references that environment. On data100, without creating a separate env first, I just ran `micromamba install -c conda-forge xeus-cling`. Then under the Launcher there were icons for C++11, C++14, and C++17.
<img width="1082" alt="Screenshot 2023-08-24 at 5 53 56 PM" src="https://github.com/berkeley-dsep-infra/datahub/assets/10248125/dc4e45c3-cdae-4c42-a65b-3fd76408ce4a">

xeus-cling could be added to biology hub's environment.yml file fairly easily.

**Comment by balajialg** [comment]
Yes, installing via micromamba works perfectly well. Thanks @ryanlovett!

Yes - we should totally add that to the environment.yml file once the faculty expresses interest in going down this route!
